game/outfits.rpy

```diff
@@ -1,289 +1,2058 @@
+                return False
+
+            elif not_reference_name is not None and self.reference_name in not_reference_name:
+                return False
+
+            elif has_accessory is not None and bool(self.has_accessory) != has_accessory:
+                return False
+
+            elif has_eyewear is not None and bool(self.has_eyewear) != has_eyewear:
+                return False
+
+            elif has_headgear is not None and bool(self.has_headgear) != has_headgear:
+                return False
+
+            elif has_necklace is not None and bool(self.has_necklace) != has_necklace:
+                return False
+
+            return True
+
+    def __register_outfit(outfit, player_created=False):
+        """
+        Registers a new outfit in the list of all outfits, allowing in-game access and persistency.
+        If the outfit has no existing corresponding persistent entry, it is saved.
+
+        IN:
+            - outfit - the JNOutfit to register.
+            - player_created - Boolean flag describing whether this outfit was defined by the current player via the new outfit creation flow
+        """
+        if outfit.reference_name in __ALL_OUTFITS:
+            jn_utils.log("Cannot register outfit name: {0}, as an outfit with that name already exists.".format(outfit.reference_name))
+
+        else:
+            if not outfit.accessory:
+                outfit.accessory = get_wearable("jn_none")
+
+            if not outfit.eyewear:
+                outfit.eyewear = get_wearable("jn_none")
+
+            if not outfit.headgear:
+                outfit.headgear = get_wearable("jn_none")
+
+            if not outfit.necklace:
+                outfit.necklace = get_wearable("jn_none")
+
+            __ALL_OUTFITS[outfit.reference_name] = outfit
+            if outfit.reference_name not in store.persistent.jn_outfit_list:
+                outfit.__save()
+
+                # If this is the first time adding it to the list, and it isn't JN or created by this player in the session, it's a new unlock
+                if not "jn_" in outfit.reference_name and not player_created:
+                    _SESSION_NEW_UNLOCKS.append(outfit)
+
+    def __register_wearable(wearable):
+        """
+        Registers a new wearable in the list of all wearables, allowing in-game access and persistency.
+        """
+        if wearable.reference_name in __ALL_WEARABLES:
+            jn_utils.log("Cannot register wearable name: {0}, as a wearable with that name already exists.".format(wearable.reference_name))
+
+        else:
+            __ALL_WEARABLES[wearable.reference_name] = wearable
+            if wearable.reference_name not in store.persistent.jn_wearable_list:
+                wearable.__save()
+
+                # If this is the first time adding it to the list, and it isn't JN, it's a new unlock
+                if not "jn_" in wearable.reference_name:
```

Comment:
Having `jn_` in name doesn't look too reliable to me. Maybe add a new prop for such flag?